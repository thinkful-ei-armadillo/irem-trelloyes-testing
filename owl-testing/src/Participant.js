import React from 'react';

function Participant(props){
    return (
        <div className="Participant">
            <img src={props.avatar} alt="headshot"/>
            <h3>{props.name}</h3>
            <p>{props.onStage}</p>
            <p>{props.id}</p>
        </div>
    );
}

export default Participant;