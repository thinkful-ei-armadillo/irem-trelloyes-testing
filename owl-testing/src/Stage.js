import React from 'react';
import Participant from './Participant';

function Stage(props){
    return (
        <section className="OnStage">
            <div className="OnStage-participants">
                {props.participants
                .filter(item => item.onStage === true)
                .map((item, index) => {
                    return <Participant key={index} name={item.name} avatar={item.avatar} />;
                    })}
            </div>
        </section>
    );

}

export default Stage;