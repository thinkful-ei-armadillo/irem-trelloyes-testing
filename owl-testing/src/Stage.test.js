import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import Stage from './Stage';
import renderer from 'react-test-renderer';


describe('<App />', () => {
    const participants=[
      {avatar: "https://ia.media-imdb.com/images/M/MV5BMTY5NzE3NzU3MF5BMl5BanBnXkFtZTgwMjg0NTQ5MDE@._V1_UX214_CR0,0,214,317_AL_.jpg",
      name: "Sally Salonga",
      onStage: "false",
      inSession: "true",
      key: '0'},
      {avatar: "https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1",
      name: "Joe Turner",
      onStage: "true",
      inSession: "true",
      key:'0'}
  ];
it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<Stage participants={participants} />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Stage participants={participants} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    })
})