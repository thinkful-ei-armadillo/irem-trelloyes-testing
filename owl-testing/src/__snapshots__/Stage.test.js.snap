// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the UI as expected 1`] = `
<section
  className="OnStage"
>
  <div
    className="OnStage-participants"
  />
</section>
`;
